### PR TITLE
Make SnackBar to be a simple function call

### PR DESCRIPTION
**Pull request by georgejor**
- Add Element dynamically to React Native root to make function call simpler
- Remove Redux and Container logic to redux complexity
- Break down props input to `title` and `options` to make `title` an essential input

### CODE REVIEW

File: src/SnackBarManager.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    this.current = null
+    this.queue = []
+  }
+
+  _hasQueue (): boolean {
+    return Array.isArray(this.queue) && this.queue.length
+  }
+
+  _addCurrent (props: SnackItemType): SnackBarManager {
+    this.current = new RootSiblings(<SnackBar {...props} onDismiss={this.dismiss} />)
```
Comment (yuripave) [inline src/SnackBarManager.js]: should be `onAutoDismiss`

File: src/SnackBarManager.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      return this._addCurrent(props)
+    }
+
+    if (isAnimated) {
+      return this
+        ._removeCurrent()
+        ._addCurrent(props)
+    }
+
+    this.current.update(<SnackBar {...props} onDismiss={this.dismiss} />)
```
Comment (yuripave) [inline src/SnackBarManager.js]: should be `onAutoDismiss`

File: src/SnackBarManager.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+  get () {
+    return {
+      current: this.current,
+      queue: this.queue
+    }
+  }
+
+  add (title: string, options?: SnackItemType): void {
+    const props = { children: title, ...options }
```
Comment (yuripave) [inline src/SnackBarManager.js]: Can we rename prop `children` to `title` to make things easier?

File: src/SnackBarManager.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+
+    if (this.current) {
+      this.queue.push(props)
+      return
+    }
+
+    this._addCurrent(props)
+  }
+
+  show (title: string, options?: SnackItemType): void {
```
Comment (yuripave) [inline src/SnackBarManager.js]: Should we expose `isAnimated` argument from `_updateCurrent`?

File: src/SnackBarManager.js (hunk 1 above)
Comment (georgejor) [inline src/SnackBarManager.js]: ✓

File: src/SnackBarManager.js (hunk 2 above)
Comment (georgejor) [inline src/SnackBarManager.js]: ✓

File: src/SnackBarManager.js (hunk 3 above)
Comment (georgejor) [inline src/SnackBarManager.js]: Good suggestion.

Comment (georgejor) [inline src/SnackBarManager.js]: ✓

File: src/SnackBarManager.js (hunk 4 above)
Comment (georgejor) [inline src/SnackBarManager.js]: As discussed, we support only slide down and show up again for nested actions. I will therefore remove the isAnimated usage.

**Pull request merged**